Sources/MapboxCoreNavigation/NavigationServiceDelegate.swift

```diff
@@ -40,6 +40,39 @@ public protocol NavigationServiceDelegate: AnyObject, UnimplementedLogging {
      */
     func navigationService(_ service: NavigationService, willRerouteFrom location: CLLocation)
     
+    /**
+     Called when navigation service has detected user taking an alternative route.
+     
+     This method is called before updating main route.
+     
+     - parameter service: The navigation service that has detected turning to the alternative.
+     - parameter route: The alternative route which will be taken as new main.
+     - parameter location: The user’s current location.
+     */
+    func navigationService(_ service: NavigationService, willTakeAlternativeRoute route: Route, at location: CLLocation?)
```

Comment:
NavigationServiceDelegate methods should be accompanied by notifications, so that different classes don’t have to compete for access to methods in the same delegate protocol.

Reply:
Added corresponding notifications